docs/cli.md

```diff
@@ -0,0 +1,158 @@
+  -s, --secrets TEXT     Secrets, either as `file=<filename>` or `env=<ENVAR>,...`; for example, `-s file=secrets.txt`
+  --name TEXT            Run name
+  --project TEXT         Project name or ID
+  -f, --func-url TEXT    Path/URL of a YAML function-configuration file, or db://<project>/<name>[:tag] for a DB function object
+  --task TEXT            Path/URL of a YAML task-configuration file
+  --handler TEXT         Invoke the function handler inside the code file
+```
+
+<a id="git-func"></a>
+## Building and Running a Function from a Git Repository
+
+To build and run a function from a Git repository, start out by adding a YAML function-configuration file in your local environment.
+This file should describe the function and define its specification.
+For example, create a **myfunc.yaml** file with the following content in your working directory:
+```yaml
+kind: job
+metadata:
+  name: remote-demo1
+  project: ''
+spec:
+  command: 'examples/training.py'
+  args: []
+  image: .mlrun/func-default-remote-demo-ps-latest
+  image_pull_policy: Always
+  build:
+    #commands: ['pip install pandas']
+    base_image: mlrun/mlrun:dev
+    source: git://github.com/mlrun/mlrun
+```
+
+Then, run the following CLI command and pass the path to your local function-configuration file as an argument to build the function's container image according to the configured requirements.
+For example, the following command builds the function using the **myfunc.yaml** file from the current directory:
+```sh
+mlrun build myfunc.yaml
+```
+
+When the build completes, you can use the `run` CLI command to run the function.
+Set the `-f` option to the path to the local function-configuration file, and pass relevant parameters.
+For example:
+```sh
+mlrun run -f myfunc.yaml -w -p p1=3
+```
+
+You can also try the following function-configuration example, which is based on the MLRun CI demo:
+```yaml
+kind: job
+metadata:
+  name: remote-git-test
+  project: default
+  tag: latest
+spec:
+  command: 'myfunc.py'
+  args: []
+  image_pull_policy: Always
+  build:
+    commands: ['pip install pandas']
+    base_image: mlrun/mlrun:dev
+    source: git://github.com/mlrun/ci-demo.git
+```
+
+<a id="sources-archive"></a>
+## Using a Sources Archive
+
+The `-a|--archive` option of the CLI [`build`](#cli-cmd-build) command enables you to define a remote object path for storing TAR archive files with all the required code dependencies.
+The remote location can be, for example, in an AWS S3 bucket or in a data container in an Iguazio Data Science Platform ("platform") cluster.
+Alternatively, you can also set the archive path by using the `MLRUN_DEFAULT_ARCHIVE` environment variable.
+When an archive path is provided, the remote builder archives the configured function sources (see the `-s|-source` [`build`](#cli-cmd-build) option) into a TAR archive file, and then extracts (untars) all of the archive files (i.e., the function sources) into the configured archive location.
```

Comment:
I'm not too familiar with this functionality, are you sure this is what it is doing ?
Why does it archive and then unarchive (extract/untar) ?

Reply:
skipped for now until we have time to better try and document the CLI